lib/core/stateMachine.js

```diff
@@ -0,0 +1,409 @@
+'use strict';
+
+var util = require('util');
+
+var Q = require('q');
+
+var getOrCreateBridge = require('./helpers/getOrCreateBridge.js');
+var originator = require('./helpers/originator.js');
+
+/**
+ * Creates a state machine for a shared extension.
+ * @param {Object} client - the asterisk client instance
+ * @param {Object} data - the data (trunks, stations, etc.) related to this
+ *   extension
+ * @param {Object} channel - the incoming channel
+ * @return {Object} stateMachine- the state machine related to this extension
```

Comment:
At least expose the {Function} stateMachine.init since it's the one that is used externally.